src/main/java/CommandManager.java

```diff
@@ -0,0 +1,268 @@
+                file.createNewFile(); // Create an empty file
+                return; // No tasks to load
+            }
+        } catch (IOException e) {
+            System.out.println("Error creating file: " + e.getMessage());
+            return;
+        }
+        if (!file.exists()) {
+            return;
+        }
+        Scanner s = new Scanner(file);
+        while (s.hasNextLine()) {
+            String line = s.nextLine();
+            Task task = parseTask(line);
+            list.add(task);
+        }
+        s.close();
+    }
+
+    private static Task parseTask(String line) {
+        char taskType = line.charAt(1);
+        boolean isDone = line.charAt(4) == 'X';
+        switch (taskType) {
+        case 'T':
+            String tDescription = line.substring(7).trim();
+            Task todo = new Todo(tDescription);
+            if (isDone) {
+                todo.markAsDone();
+            }
+            return todo;
+        case 'D':
+            int byIndex = line.indexOf("(by:");
+            String dDescription = line.substring(7, byIndex).trim();
+            String deadline = line.substring(byIndex + 4, line.length() - 1).trim();
+            Task deadlineTask = new Deadline(dDescription, deadline);
+            if (isDone) {
+                deadlineTask.markAsDone();
+            }
+            return deadlineTask;
+        case 'E':
+            int fromIndex = line.indexOf("(from:");
+            int toIndex = line.indexOf(", to:");
+            String eDescription = line.substring(7, fromIndex).trim();
+            String startTime = line.substring(fromIndex + 6, toIndex).trim();
+            String endTime = line.substring(toIndex + 5, line.length() - 1).trim();
+            Task eventTask = new Event(eDescription, startTime, endTime);
+            if (isDone) {
+                eventTask.markAsDone();
+            }
+            return eventTask;
+        default:
+            return null;
+        }
+    }
+
+    public static void printHoriLine() {
+        System.out.println("__________________________________________________");
+    }
+
+    public static void printList(ArrayList<Task> list) {
+        printHoriLine();
+        if (list.isEmpty()) {
+            indentMessage("No tasks found!");
+        } else {
+            for (int i = 0; i < list.size(); i++) {
+                indentMessage((i + 1) + ": " + list.get(i).toString());
+            }
+        }
+        printHoriLine();
+    }
+
+    public static void printNewEntry(ArrayList<Task> list, Task key) {
+        printHoriLine();
+        indentMessage("Added task to list:");
+        indentMessage(key.toString());
+        indentMessage("Now you have " + list.size() + " task(s)");
+        printHoriLine();
+    }
+
+    private static void markTask(ArrayList<Task> list, String text, boolean done) {
+        String number = text.replaceAll("\\D+", ""); // Remove all non-digits
+        int i = Integer.parseInt(number);
```

Comment:
The name `i` is not intuitive, maybe call it `taskNumber` instead. Additionally, the variable `number` can be called `parsedNumber` instead.